docs/attributes-registry/process.md

```diff
@@ -11,12 +11,32 @@
 | `process.command` | string | The command used to launch the process (i.e. the command name). On Linux based systems, can be set to the zeroth string in `proc/[pid]/cmdline`. On Windows, can be set to the first parameter extracted from `GetCommandLineW`. | `cmd/otelcol` |
 | `process.command_args` | string[] | All the command arguments (including the command/executable itself) as received by the process. On Linux-based systems (and some other Unixoid systems supporting procfs), can be set according to the list of null-delimited strings extracted from `proc/[pid]/cmdline`. For libc-based executables, this would be the full argv vector passed to `main`. | `[cmd/otecol, --config=config.yaml]` |
 | `process.command_line` | string | The full command used to launch the process as a single string representing the full command. On Windows, can be set to the result of `GetCommandLineW`. Do not set this if you have to assemble it just for monitoring; use `process.command_args` instead. | `C:\cmd\otecol --config="my directory\config.yaml"` |
+| `process.end` | string | The date and time the process ended, in ISO 8601 format. | `2023-11-21T09:26:12.315Z` |
+| `process.env_vars` | string[] | Array of environment variable bindings. [1] | `[PATH=/usr/local/bin;/usr/bin, USER=ubuntu]` |
```

Comment:
I see this is marked as `beta` in ECS. Does any of you know if it was intended to take all (even beta) fields from ECS? Are those in use by ECS customers/users? Without knowing what it is used today, it's kinda hard to see any value.. specially given SDKs may need to even do extra things to protect the value from it.

Reply:
This field is being used by Elastic users now. One example of a current use case is that there are security detections on the use of `LD_PRELOAD` and `LD_LIBRARY_PATH`.

I'm not sure right now if the ECS merger should take beta fields, I'll try to find out

Reply:
Hi @joaopgrassi, looking at the [OTEP to merge ECS](https://github.com/open-telemetry/oteps/blob/main/text/0199-support-elastic-common-schema-in-opentelemetry.md), it says `OTel SemConv will adopt ECS in its full scope (except for individual adjustments in detail where inevitable)`, so maybe beta fields can be taken, I'm not sure, it's a bit vague about what "full scope", or "individual adjustments" means. I couldn't find anything referencing beta fields specifically.

For this particular field, `process.env_vars` will be promoted from beta to GA in the next ECS release https://github.com/elastic/ecs/pull/2315

Reply:
I just stumbled upon this while searching for a standardized way to store (specific) environment variables in resource attributes. To be honest I am surprised about the suggested format:

```
[PATH=/usr/local/bin;/usr/bin, USER=ubuntu]`
```

Similar to HTTP headers I would rather have expected a format like

```
| `process.environment_variable.<key>` | string | Process environment variables, `<key>` being the environment variable name, the value being the environment variable value. | `proccess.environment_variable.PATH="/usr/local/bin;/usr/bin"`; `process.environment_variable.USER="ubuntu"` | Opt-In |
```

This makes it much easier to (a) find environment variables in the backend (or collector) and (b) only collect those environment variables I really care about (and to create filters to remove those not wanted)

Reply:
+1 to this suggestion for consistency.

Reply:
Hi @svrnm, thanks for the suggestion. I think this will need more discussion, so I've taken `env_vars` out of this PR, and created an [issue](https://github.com/open-telemetry/semantic-conventions/issues/672) to decide how to implement this before it's added in the future

Reply:
thanks @mjwolf , that totally makes sense to have this in a separate discussion, I will follow up there